p2p/protocol/circuitv1/relay/relay.go

```diff
@@ -72,19 +86,33 @@ func NewRelay(h host.Host, opts ...Option) (*Relay, error) {
 func (r *Relay) Close() error {
 	if atomic.CompareAndSwapInt32(&r.closed, 0, 1) {
 		r.host.RemoveStreamHandler(ProtoID)
+		r.scope.Done()
 		r.cancel()
 	}
 	return nil
 }
 
 func (r *Relay) handleStream(s network.Stream) {
-	s.SetReadDeadline(time.Now().Add(StreamTimeout))
-
 	log.Debugf("new relay stream from: %s", s.Conn().RemotePeer())
 
+	if err := s.Scope().SetService(ServiceName); err != nil {
+		log.Debugf("error attaching stream to relay service: %s", err)
+		s.Reset()
+		return
+	}
+
+	if err := s.Scope().ReserveMemory(maxMessageSize, network.ReservationPriorityAlways); err != nil {
```

Comment:
Are we sure that all relay usages will have the highest priority? This seems off... I would expect nodes to offer relay services on best-effort basis, except for dedicated relay supernodes.

Reply:
highest priority means it wont work if you can't allocate this buffer; it is fine to fail!

Reply:
note that there are tight limits everywhere at play, so this will get throttled early at the service or peer level.
I would find it unlikely to hit the system level.